app/views/compute_resources_vms/form/libvirt/_base.html.erb

```diff
@@ -5,6 +5,11 @@
 </div>
 <%= select_f f, :cpu_mode, Foreman::Model::Libvirt::CPU_MODES, :to_s, :to_s, { }, :label => _("CPU mode"), :label_size => "col-md-2" %>
 <%= byte_size_f(f, :memory, disabled: !new_vm, label: _('Memory'), recommended_max_value: compute_resource.max_memory.to_i) %>
+<%= field(f, :firmware, :disabled => !new_vm, :label => _('Firmware'), :label_size => "col-md-2") do
+  compute_resource.firmware_types.collect do |type, name|
+    radio_button_f f, :firmware, {:disabled => !new_vm, :value => type, :text => _(name)}
```

Comment:
Shouldn't it be:
```suggestion
<%= field(f, :os_firmware, :disabled => !new_vm, :label => _('Firmware'), :label_size => "col-md-2") do
  compute_resource.firmware_types.collect do |type, name|
    radio_button_f f, :os_firmware, {:disabled => !new_vm, :value => type, :text => _(name)}
```

Reply:
No, the controller accepts firmware (just like the VMware side). The libvirt model then converts the option to what fog-libvirt accepts.

Reply:
Why do we need to consider VMware here? I don't see the necessity for VMware and Libvirt forms to have the same naming convention.
If we keep it as it is, we encounter the following error: `Failure: undefined method firmware for <Fog::Libvirt::Compute::Server` ...